storage_service/locations/models/pipeline.py

```diff
@@ -126,3 +128,76 @@ def create_default_locations(self, shared_path=None):
                     p['purpose'], location, self)
                 LocationPipeline.objects.get_or_create(
                     pipeline=self, location=location)
+
+    # HTTP API CALLS
+
+    def _request_api(self, method, path, fields=None):
+        url = 'http://{}/api/{}'.format(
+            self.remote_name,
+            path.lstrip('/'),
+        )
+        data = {
+            'username': self.api_username,
+            'api_key': self.api_key
+        }
+        if fields:
+            data.update(fields)
+
+        LOGGER.debug('URL: %s; data: %s', url, data)
+        try:
+            resp = requests.request(method, url, data=data, allow_redirects=True)
+        except requests.exceptions.RequestException, e:
+            LOGGER.error('Unable to connect to pipeline %s.', self)
+            _, _, tb = sys.exc_info()
+            raise PipelineClientException, PipelineClientException(e), tb
+        else:
+            LOGGER.debug('Response: %s %s', resp.status_code, resp.text)
+            return resp
+
+    def get_processing_config(self, name):
+        """
+        Obtain a processing configuration XML document given its name. The
+        content is returned as a string, which may be empty in case of errors.
+        """
+        if not name:
+            raise ValueError('Parameter name is empty.')
+        url = 'processing-configuration/' + name
+        resp = self._request_api('GET', url)
+        if resp.status_code != requests.codes.ok:
+            raise PipelineClientUnexpectedCodeException(self, resp.status_code)
+        if not resp.text:
+            raise PipelineClientException('Pipeline {}: empty processing configuration ({}).'.format(self, name))
```

Comment:
This or the docstring should be updated to reflect the actual empty config behaviour.